python/lsst/daf/butler/datastores/webdavDatastore.py

```diff
@@ -0,0 +1,271 @@
+# This file is part of daf_butler.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+"""Webdav datastore."""
+
+__all__ = ("WebdavDatastore", )
+
+import logging
+import os
+import pathlib
+import tempfile
+
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Optional,
+    Type,
+    Union,
+)
+
+from lsst.daf.butler import (
+    ButlerURI,
+    DatasetRef,
+    Formatter,
+    Location,
+    StoredFileInfo,
+)
+
+from .fileLikeDatastore import FileLikeDatastore
+
+from lsst.daf.butler.core.webdavutils import (
+    getHttpSession,
+    folderExists,
+    webdavCheckFileExists,
+    webdavDeleteFile,
+)
+
+if TYPE_CHECKING:
+    from .fileLikeDatastore import DatastoreFileGetInformation
+    from lsst.daf.butler import DatastoreConfig
+    from lsst.daf.butler.registry.interfaces import DatastoreRegistryBridgeManager
+
+log = logging.getLogger(__name__)
+
+
+class WebdavDatastore(FileLikeDatastore):
+    """Basic Webdav Storage backed Datastore.
+
+    Parameters
+    ----------
+    config : `DatastoreConfig` or `str`
+        Configuration. A string should refer to the name of the config file.
+    bridgeManager : `DatastoreRegistryBridgeManager`
+        Object that manages the interface between `Registry` and datastores.
+    butlerRoot : `str`, optional
+        New datastore root to use to override the configuration value.
+
+    Raises
+    ------
+    ValueError
+        If root location does not exist and ``create`` is `False` in the
+        configuration.
+
+    Notes
+    -----
+    """
+
+    defaultConfigFile = "datastores/webdavDatastore.yaml"
+    """Path to configuration defaults. Relative to $DAF_BUTLER_DIR/config or
+    absolute path. Can be None if no defaults specified.
+    """
+
+    def __init__(self, config: Union[DatastoreConfig, str],
+                 bridgeManager: DatastoreRegistryBridgeManager, butlerRoot: str = None):
+        super().__init__(config, bridgeManager, butlerRoot)
+        b = ButlerURI(self.root)
```

Comment:
You probably want a `forceDirectory=True` here.  I need to clean up all this to ensure that self.root is always a ButlerURI.

Also, `b` means nothing to the reader so can you say `uri` as the variable instead?